cpp/models/abm/person.h

```diff
@@ -70,127 +49,141 @@ class Person
 {
 public:
     /**
-     * create a Person.
-     * @param id index and type of the initial location of the person
-     * @param infection_properties the initial infection state of the person and if infection is detected
-     * @param vaccination_state the initial infection state of the person
-     * @param age the age group of the person
-     * @param global_params the global infection parameters
-     * @param person_id index of the person
+     * @brief Create a Person.
+     * @param location Initial location of the person.
+     * @param age The age group of the person.
+     * @param person_id Index of the person.
      */
-    Person(LocationId id, InfectionProperties infection_properties, AgeGroup age,
-           const GlobalInfectionParameters& global_params,
-           VaccinationState vaccination_state = VaccinationState::Unvaccinated, uint32_t person_id = INVALID_PERSON_ID);
+    explicit Person(Location& location, AgeGroup age, uint32_t person_id = INVALID_PERSON_ID);
 
     /**
-     * create a Person.
-     * @param location the initial location of the person
-     * @param infection_properties the initial infection state of the person and if infection is detected
-     * @param age the age group of the person
-     * @param global_params the global infection parameters
-     * @param person_id index of the person
-     */
-    Person(Location& location, InfectionProperties infection_properties, AgeGroup age,
-           const GlobalInfectionParameters& global_params,
-           VaccinationState vaccination_state = VaccinationState::Unvaccinated, uint32_t person_id = INVALID_PERSON_ID);
+    * compare two persons
+    */
+    bool operator==(const Person& other) const
+    {
+        return (m_person_id == other.m_person_id);
+    }
 
     /** 
-     * Time passes and the person interacts with the population at its current location.
-     * The person might change infection state.
-     * @param dt length of the current simulation time step
-     * @param global_infection_parameters infection parameters that are the same in all locations
+     * @brief Time passes and the person interacts with the population at its current location.
+     * The person might become infected.
+     * @param[in] t Current time.
+     * @param[in] dt Length of the current simulation time step.
+     * @param[in,out] global_infection_parameters Infection parameters that are the same in all locations.
      */
-    void interact(TimeSpan dt, const GlobalInfectionParameters& global_infection_parameters, Location& loc);
+    void interact(TimePoint t, TimeSpan dt, GlobalInfectionParameters& params);
```

Comment:
but they aren't drawn in `interact`, are they? aren't they used the same as in the ODE simulation, draw the parameters before running the simulation?

Reply:
They are drawn in each `Infection` constructor and therefore possibly in `interact`. The problem is that `get_sample` in the `ParameterDistributions` is not const due to the predefined samples.

Reply:
makes sense. Actually, they are also not const because they used to have their own local random number generator as well. Maybe the parameters should only store the parameters of the distributions then instead of `ParameterDistributionUniform` objects.

Reply:
We can do that. I will prepare that for the next update, probably next week. Then we can have them const.
Similar to how it is done in `random_events`, one could then create a `ParameterDistributionUniform` each time a draw is needed. However, as these parameters are const, it might make sense to have them only once, maybe through a static member variable of Infection/ViralLoad respectively?

Reply:
no, I think it should be possible to have multiple simulations at the same time and try different parameters.

Reply:
Now uses const Parameters, after the implementation changes in the DistributionAdapters.